src/ducks/context/SelectionContext.jsx

```diff
@@ -0,0 +1,70 @@
+import React, {
+  createContext,
+  useState,
+  useCallback,
+  useEffect,
+  useContext
+} from 'react'
+
+import flag from 'cozy-flags'
+
+export const SelectionContext = createContext()
+
+export const useSelectionContext = () => {
+  return useContext(SelectionContext)
+}
+
+const SelectionProvider = ({ children }) => {
+  const [isSelectionModeActive, setIsSelectionModeActive] = useState(false)
+  const [selected, setSelected] = useState([])
+  const isSelectionModeEnabled = flag('banks.selectionMode.enabled')
+
+  const activateSelectionMode = useCallback(
+    () => isSelectionModeEnabled && setIsSelectionModeActive(true),
+    [isSelectionModeEnabled]
+  )
+
+  const deactivateSelectionMode = () => setIsSelectionModeActive(false)
+
+  const addToSelection = item => {
+    setSelected(v => [...v, item])
+  }
+
+  const removeFromSelection = useCallback(
+    item => {
+      setSelected(selected.filter(e => e.id !== item.id))
+    },
+    [selected]
+  )
+
+  const emptySelection = () => setSelected([])
+
+  const isSelected = useCallback(item => selected.includes(item), [selected])
+
+  useEffect(() => {
+    if (isSelectionModeActive && selected.length === 0) {
+      deactivateSelectionMode()
+    }
+    if (!isSelectionModeActive && selected.length > 0) {
+      activateSelectionMode()
+    }
+  }, [selected, activateSelectionMode, isSelectionModeActive])
+
+  return (
+    <SelectionContext.Provider
+      value={{
+        isSelectionModeActive,
+        selected,
+        addToSelection,
+        isSelected,
+        emptySelection,
+        removeFromSelection,
+        isSelectionModeEnabled
+      }}
```

Comment:
I'd memo the value passed to context to prevent unnecessary renders.

Reply:
genre quelque chose comme ça ?
```
const value = useMemo(() => ({isSelectionModeActive, ...}))
<SelectionContext.Provider value={value}>
```

Reply:
Oui.

Reply:
Ok compris